### DISCUSSION TITLE
music-assistant crashing, not producing enough logs to get to the bottom of things...

### DISCUSSION THREAD

**Discussion by terribletrouble** [music-assistant/support#2731] [Q&A] [answered]
#0 - love music-assistant - well done!

#1 - the discord server - does not seem to work, failed to load messages.

#2 - docker env. stand alone, not integrated into HA - maybe this problem below is the reason. tried the integrated version first.
scanned library via smb, took about 3 or 4 days for the blue spinning circle to dissapear.
I can now play about 30% of one album, then things just stop.
Only way to recover is to 'docker-compose restart' - running beta version

> 2024-08-07 10:16:03.815 INFO (MainThread) [music_assistant] Starting Music Assistant Server (20b4eb0a512b42999a18bfaed2bd78c8) version 2.2.0b3 - HA add-on: False - Safe mode: False

These are the best logs I can get - I cannot find how to enable verbose or debug logging....
This is after restarting MA, then playing one album and having things stop:

`
![image](https://github.com/user-attachments/assets/1bbd2007-4600-4835-a359-ceaabc609611)
`

**Comment by OzGav** [comment]
#1. Did you press on the thumbs up in the verified channel to get access?

#2 looking at you logs you have many tagging related errors https://music-assistant.io/faq/troubleshooting/

**Reply by OzGav** [reply]
Logs are available in MA SETTINGS>>CORE. Please don’t enable debug or verbose unless asked. Also only ATTACH the logs, we don’t want to scroll through pages here. How big is the music library you are trying to import?

**Reply by terribletrouble** [reply]
These logs, just do not produce something that helps me to investigate:
music-assistant    | 2024-08-15 15:25:26.089 WARNING (MainThread) [music_assistant.music.track] Unable to fetch artist details library://artist/158 - artist not found in library: 158
music-assistant    | 2024-08-15 15:25:26.096 WARNING (MainThread) [music_assistant.music.track] Unable to fetch artist details library://artist/158 - artist not found in library: 158
music-assistant    | 2024-08-15 15:25:26.098 WARNING (MainThread) [music_assistant.music.track] Unable to fetch artist details library://artist/158 - artist not found in library: 158

**Reply by OzGav** [reply]
That looks like you have a DB problem as well. Have you removed a provider (Spotify maybe?) or tried to edit the DB yourself?

**Reply by terribletrouble** [reply]
I have not, but after our last communication, I deleted everything - then started fresh, I switch my docker to the Beta, and at one stage saw a migration of the DB from 3 to 5, that took about 3 or 4 days.
My total file count of mp3 & flac is 152123.
I know all the tagging is not perfect, it never is.

**Reply by OzGav** [reply]
So this can be closed then?

**Reply by marcelveldt** [reply]
Why is it so slow? What kind of hardware are you using and does MA have direct access to the files or is it with smb?

**Reply by terribletrouble** [reply]
Well it is still not getting through my whole collection, seems to stop at around 130 thousand and does not go any more. Now, starting all again, on the current Beta, many many logs about missing albumartist. But no status as to how it doing, how many tracks processed, but it is still looking very slow.

**Reply by terribletrouble** [reply]
@marcelveldt all with SMB, files on my NAS, MA running as an LXC on proxmox, running on SSD, with 8Gb and dual cpu's.

**Reply by OzGav** [reply]
MA is probably not going to end up working well if you haven’t set the album artist correctly.

**Reply by marcelveldt** [reply, marked as answer]
It will work without album artist but super slow as it has to hunt for details all the time

**Reply by terribletrouble** [reply]
Thanks guys, that is probably the answer that will help explain things. I do think you should relax things. Whose music collection is perfect? I have old weird albums that when you run through picard/musicbrainz, they break into multiple albums and just do things incorrectly, but I own the original CD, so I know what is right. 
I was very impressed with Navidrome, I am sure you know it - but it has many short commings, cannot play on remote systems and does not integrate into HA, but from a pure library/access point of view, it brushes over inconsistencies in tagging and is fast.
The album artist is also a very difficult tag to set correctly, often not something you can just automate.

**Reply by marcelveldt** [reply]
Music Assistant is mainly used for streaming providers and small filebased libraries. Really large file collections are rare, often because people most often already have settled on a better solution for that, such as Navidrome, Roon, LMS etc.

We have on our radar to improve the filebased provider speed but its not the highest priority as we have so many other open tasks as well and limited capacity. I made several changes to our model so we can actually handle super large libraries ( I tested with a 3 million tracks test db) but making the scanner faster is a large task. We make small improvements with each iteration but it will probably take us months if not years before we are on par with other (commercial) products. Its on the radar and we know; just not enough hours in a day to fix it.
